### PR TITLE
Headers added to csv output files

### PR DESCRIPTION
**Pull request by sadrasabouri**
#### Reference Issues/PRs
#282 
#### What does this implement/fix? Explain your changes.
+ add `header` flag to `save_csv` method
+ add `header` option to `csv_matrix_print` method

### CODE REVIEW

File: pycm/pycm_obj.py [hunk 1]
```diff
@@ -398,7 +401,8 @@ def save_csv(
                 if normalize:
                     matrix = self.normalized_table
                 csv_matrix_file = open(name + "_matrix" + ".csv", "w")
```
Comment (sepandhaghighi) [inline pycm/pycm_obj.py]: Add `encoding="utf-8"`
**Note** : Reason of `appveyor` failure

File: pycm/pycm_output.py [hunk 2]
```diff
@@ -353,22 +353,32 @@ def sparse_table_print(sparse_matrix):
...
+    :param header: add headers to .csv file
+    :type header: bool
     :return:
     """
     result = ""
+    header_section = ""
     classes.sort()
     for i in classes:
+        if header is True:
+            header_section += i
```
Comment (sepandhaghighi) [inline pycm/pycm_output.py]: header_section += `str(i)`

**Example** : `cm = ConfusionMatrix([1,2,3,4],[1,2,3,4])`

File: pycm/pycm_output.py [hunk 3]
```diff
@@ -353,22 +353,32 @@ def sparse_table_print(sparse_matrix):
...
+            if i != classes[-1]:
```
Comment (sepandhaghighi) [inline pycm/pycm_output.py]: Logic is right, but this condition will check in **each iteration**, I think it should be check at the end of this function (only a single if) and out of any loop ;-)

Comment (sadrasabouri) [reply pycm/pycm_output.py]: Yeah i though about this alternative, i thought this approach won't make header section in the case that we don't need it.
but as you say i'll change it.

Comment (sepandhaghighi) [reply pycm/pycm_output.py]: > Yeah i though about this alternative, i thought this approach won't make header section in the case that we don't need it.
> but as you say i'll change it.

I only mentioned second `if ... else` (adding `,` and `n`), the first `if` position was true and you are right about it ;-)

File: pycm/pycm_output.py [hunk 4]
```diff
@@ -353,22 +353,32 @@ def sparse_table_print(sparse_matrix):
...
     :type table:dict
+    :param header: add headers to .csv file
+    :type header: bool
     :return:
     """
     result = ""
+    header_section = ""
     classes.sort()
     for i in classes:
+        header_section += str(i)
```
Comment (sepandhaghighi) [inline pycm/pycm_output.py]: In previous review I only mentioned second `if ... else` (adding`,` and `n`), the first `if` (creating header_section) position was true

File: Test/output_test.py [hunk 5]
```diff
@@ -491,6 +491,12 @@
 >>> save_csv_data = cm.save_csv("test")
 >>> save_csv_data["Status"]
 True
+>>> save_csv_data = cm.save_csv("test", header=False)
+>>> save_csv_data["Status"]
+True
+>>> save_csv_data = cm.save_csv("test", header=True)
+>>> save_csv_data["Status"]
+True
```
Comment (sepandhaghighi) [inline Test/output_test.py]: Also add this test  : 

```pycon
>>> save_csv_data = cm.save_csv("test_header", header=True,matrix_save=True)
>>> save_csv_data["Status"]
True
>>> os.remove("test_header.csv")
>>> os.remove("test_header_matrix.csv") 
```

File: pycm/pycm_output.py (hunk 4 above)
Comment (sadrasabouri) [reply pycm/pycm_output.py]: Alright, i fixed it.
There is a double check of `header` at the end of function due to it's right functionality for the cases that we do not need header, so last if (`if len(header_section > 0):`) is necessary i think.

**Comment by sepandhaghighi** [comment]
@alirezazolanvari Please review this PR ;-)

**Comment by sepandhaghighi** [comment]
I made some minor edits, now LGTM 👍 
@sadrasabouri Well done 💯

**Comment by sadrasabouri** [comment]
> I made some minor edits, now LGTM +1
> @sadrasabouri Well done 100

Feel free.

**Pull request merged**